### ISSUE TITLE
Handling upgrades that don't apply 

### ISSUE THREAD

**Issue by martbhell** [CSC-IT-Center-for-Science/ansible-role-dell-firmware-upgrade#25]
While I remember.

On Friday we saw some disk firmwares available. Disks are behind an Integrated Controller and when dsu applies the upgrade it can't because of System Configuration or some such. This role is green. 

Would be nice if this role could alert somehow about this. Don't have return code right now but will try to find out :)

**Comment by Klaas-** [comment]
I don't think that's an exit code issue in this role. If it didn't throw a failure it exited properly and had one of these exit codes: https://github.com/CSCfi/ansible-role-dell-firmware-upgrade/blob/master/defaults/main.yml#L30-L33 none of these indicate a partial failure, those exit codes would be 25/26.

I always debug the command_result of this role to show the stdout, that's helpful when reporting errors and finding failures :)

For reference: latest docs: https://topics-cdn.dell.com/pdf/system-update-v17_users-guide_en-us.pdf

**Comment by martbhell** [comment]
Indeed. It might be that we need to match / fail if there is something in stdout for this. But maybe instead of adding this to the role it should be reported to Dell.

**Comment by Klaas-** [comment]
yeah, that's what I meant, if the disk firmware upgrade was unsuccessful it should give a different exit code --> ie issue in dsu, not this role.

**Comment by martbhell** [comment]
E-mailed dell's linux poweredge mailing list let's see if it goes through the mailing list. Return Code is zero.
Is that a good enough place or have you @Klaas-  had better success with contacting dell tech support?

<pre>
This was tested on a CentOS 7.6 Dell PowerEdge C6320 server.

output from dsu:


# dsu --non-interactive --apply-upgrades; echo $?
DELL EMC System Update 1.7.0
Copyright (C) 2014 DELL EMC Proprietary.
Verifying catalog installation ...
Installing catalog from repository ...
Fetching dsucatalog ...
Reading the catalog ...
Verifying inventory collector installation ...
Getting System Inventory ...
warning: Inventory collector returned with partial failure.
Determining Applicable Updates ...
Fetching Serial-ATA_Firmware_7KW56_LN_AA65_A00 ...
Installing Serial-ATA_Firmware_7KW56_LN_AA65_A00
Collecting inventory...
....
Running validation...

This Update Package is not compatible with your system configuration.

Fetching Serial-ATA_Firmware_KVKY5_LN_NA07_A00 ...
Installing Serial-ATA_Firmware_KVKY5_LN_NA07_A00
Collecting inventory...
....
Running validation...

This Update Package is not compatible with your system configuration.

Done! Please run 'dsu --inventory' to check the inventory
Exiting DSU!
0

</pre>

**Comment by Klaas-** [comment]
All the mailing lists are down because of a misconfigured mailserver or something like that, I've complained multiple times in past, don't think that changed yet... :) So no hope there.

For support cases: I've only had bad dell support experience when it comes to software issues, so I can't point to any helpful contacts for dsu.  So in short: open a support case, don't hope for a fast resolution :)

**Comment by martbhell** [comment]
> All the mailing lists are down because of a misconfigured mailserver or something like that, I've complained multiple times in past, don't think that changed yet... :) So no hope there.
> 
> For support cases: I've only had bad dell support experience when it comes to software issues, so I can't point to any helpful contacts for dsu. So in short: open a support case, don't hope for a fast resolution :)

Thanks, I'm glad I wrote about this here or I would have waited a while!

**Comment by Klaas-** [comment]
do you also have rhel servers to test on? While I do agree that centos and rhel are functionally equal if some package actually tests for rhel artifacts it may not find them. "Officially" Dell only supports rhel afaik.

**Comment by martbhell** [comment]
I do not. I guess we'll see what happens. I couldn't find anything about this in the firmware binary but I didn't look very far :)

**Comment by Klaas-** [comment]
It seems the poweredge mailing lists just came back online -- maybe it would be a good idea to also report it there if you haven't gotten help from support yet :)

**Comment by martbhell** [comment]
Great news! We could try there too, I'm on holiday now so 'not me' :)
We have not heard back from Dell DSU team but they have been informed.

**Comment by martbhell** [comment]
No news yet but I tried to e-mail the linunx poweredge mailing list too to spread the Word
